### ISSUE TITLE
prisma studio crashes when connecting to Supabase with: 'Segmentation fault' error

### ISSUE THREAD

**Issue by jm61** [prisma/prisma#14122]
### Bug description

When starting prisma studio, logs in the console are okay until message 'Prisma Studio is up on http://localhost:5555' then crashes with msg: ' ERROR  Command was killed with SIGSEGV (Segmentation fault): /home/xxx/.local/share/pnpm/store/v3/tmp/dlx-301926/node_modules/.bin/prisma studio

### Expected behavior

to see prisma studio interface when connecting to Supabase

### Prisma information

I followed this guide: https://supabase.com/docs/guides/integrations/prisma
the prisma migrate running well, tables are created on Supabase

### Environment & setup

Linux Mint 20.3
node v17.3.1
pnpm v7.3.0

### Prisma Version

```

```
V4.0

**Comment by ibreslauer** [comment]
I'm having the same issue connecting to Supabase, both when starting Prisma Studio or querying the db. Doesn't happen when connecting to, for example, Heroku.

```
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Prisma Studio is up on http://localhost:5555
Segmentation fault
```

**Comment by janpio** [comment]
What is your node version @ibreslauer?

**Comment by janpio** [comment]
If you tell us your complete node version where this was failing, we could try to repoduce it and maybe even fix it. Without that information, no.

**Comment by ibreslauer** [comment]
@janpio v18.7.0

**Comment by aqrln** [comment]
Explanation: https://github.com/prisma/prisma/issues/10649#issuecomment-1247961614

**Comment by aqrln** [comment]
Thank you for the report!

I'm closing this issue in favour of https://github.com/prisma/prisma/issues/10649 as a duplicate to consolidate them and make it easier to keep track of the issue. Please post any questions or additional details in that thread, or open a new issue if you experience a similar problem under completely different conditions (e.g., with other DBMS or operating systems).

While we are looking for the best way to fix or prevent the underlying problem, the recommended workaround for affected users is to make sure that the versions of OpenSSL used by Node.js and the system OpenSSL match:
* either by downgrading Node.js to 16.x,
* or by upgrading the operating system (or Docker image) to use OpenSSL 3 — this might be less suitable for production systems, but this is a good option if you only encounter the issue on you local machine during development.

Alternatively, if you can't do that and you need to stay with Node.js 18 on a system with OpenSSL 1.1.x, you can switch to the [binary engine](https://www.prisma.io/docs/concepts/components/prisma-engines/query-engine#defining-the-query-engine-type-for-prisma-client) in Prisma.

**Comment by aqrln** [comment]
This issue is fixed in **Prisma 4.13.0** which will be released tomorrow, on **Tuesday, Apr 18**. If you need to temporarily use a pre-release version until then, you can use `4.13.0-dev.33`.

---

For future readers:

Note that this GitHub issue is about a specific bug that occurs when using **Node.js 17+ on a system with OpenSSL 1.1 on x86-64 CPUs when using TLS to connect to database**. Please do not post comments here if you encounter a segfault under different circumstances, and **open a new issue instead**.

If you are facing segfaults on ARM, please leave a comment in https://github.com/prisma/prisma/issues/18510 or open a new issue.
